drivers/perf/raspberrypi_axi_monitor.c

```diff
@@ -515,17 +699,35 @@ static const struct file_operations fops_debug = {
 	.open = simple_open
 };
 
+static const struct of_device_id rpi_axiperf_match[] = {
+	{ .compatible = "brcm,bcm2835-axiperf",
+	  .data = &config_2835 },
+	{ .compatible = "brcm,bcm2711-axiperf",
+	  .data = &config_2711 },
+	{ .compatible = "brcm,bcm2712-axiperf",
+	  .data = &config_2712 },
+	{},
+};
+MODULE_DEVICE_TABLE(of, rpi_axiperf_match);
+
 static int rpi_axiperf_probe(struct platform_device *pdev)
 {
 	int ret = 0, i;
 	struct device *dev = &pdev->dev;
 	struct device_node *np = dev->of_node;
 	struct device_node *fw_node;
+	const struct of_device_id *match;
 
 	state = kzalloc(sizeof(struct rpi_axiperf), GFP_KERNEL);
 	if (!state)
 		return -ENOMEM;
 
+	match = of_match_device(rpi_axiperf_match, dev);
```

Comment:
You can use `match = of_device_get_match_data(dev);` to save having to move `rpi_axiperf_match`.